spec/requests/api/v1/participant_declarations_spec.rb

```diff
@@ -14,29 +14,17 @@
       {
         participant_id: ect_profile.user.id,
         declaration_type: "started",
-        declaration_date: (Time.zone.now - 1.week).iso8601,
+        declaration_date: (ect_profile.schedule.milestones.first.start_date + 1.day).rfc3339,
```

Comment:
why are we changing from `iso8601` to `rfc3339`? there are other places using `iso8601` in the API. we should probably change these globally for our api. otherwise some endpoints need `iso8601` and others `rfc3339` which as an api consumer is a bad experience

Reply:
The reason I changed it here is because `(ect_profile.schedule.milestones.first.start_date + 1.day).iso8601` does not work - presumably it does not include the time part, which `rfc3339` does. If I use `(ect_profile.schedule.milestones.first.start_date + 24.hours). iso8601` it works.

The larger point of 'we should use the same format across the API'. Yeah. We probably should. Something to improve in another PR.

Reply:
if you did not explicitly want to change the API, i recommend `(ect_profile.schedule.milestones.first.start_date + 24.hours). iso8601` over the current change. as with this change we are not explicitly saying the client should now send us `rfc3339` and we are testing for that and not `iso8601`.

If you did want to change to `rfc3339` the API the docs at https://ecf-sandbox.london.cloudapps.digital/lead-providers/guidance/reference#post-api-v1-participant-declarations will need updating. But i'd like to also know why we decided to change.

Reply:
rfc3339 is a profile of iso8601. All rfc3339 is iso8601, but not the other way around

Reply:
The docs say 
```
string, ISO 8601 date with time and timezone
```
ISO without time and timezone is invalid according to them, although I guess we could say `rfc3339` in there.

As for why, well, we care about the exact time on declarations. I think in general this is not true in all of our endpoints - things like date of birth won't require this degree of precision.

Reply:
We specify ISO 8601 date with time and timezone, which is what RFC3339 defines as a stricter parsing format. From the Ruby Docs:

"If the time or zone components are missing then an ArgumentError will be raised. This is much stricter than either parse or iso8601 which allow for missing components."

As we don't want to allow missing components, this is what is defined by RFC3339.

Reply:
I'm happy to moving to `rfc3339`. I don't believe they are quite super/sub sets of one another. and there are subtle differences. here is a naive example for the ruby implementation
```
irb(main):030:0> DateTime.iso8601("20-12-09T16:09:53+00:00")
=> Wed, 09 Dec 2020 16:09:53 +0000
irb(main):031:0> DateTime.rfc3339("20-12-09T16:09:53+00:00")
Traceback (most recent call last):
        2: from (irb):31
        1: from (irb):31:in `rfc3339'
Date::Error (invalid date)
```
So I think we just want to update the docs to reflect that and remove any ambiguity. Happy for this to be in another PR

Reply:
The former is an invalid date since it doesn't say what the offset it being applied to. For the full representation it needs to specify offset from a specific timezone. It's entirely applicable for that offset to be from localtime, but the Z forces it to be from UTC. Hence it's unambiguous adoption.